processors/filterprocessor/fields/filter.go

```diff
@@ -0,0 +1,72 @@
+package fields
+
+import (
+	"fmt"
+
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/indexer/data"
+)
+
+// Operation an operation like "some" or "all" for boolean logic
+type Operation string
+
+const someFieldOperation Operation = "some"
+const allFieldOperation Operation = "all"
```

Comment:
nit: should we name these Ops Or, And?

Reply:
We can.  Would it be more clear to devs?

Reply:
I think it would since and/or are commonly used operations. `matchall` and `matchany` would also be more clear.